packages/core/src/agent/index.ts

```diff
@@ -670,17 +674,19 @@ export class Agent<
     messages: CoreMessage[];
     runtimeContext: RuntimeContext;
   }) {
+    const instructionStringOrFunction = instructionsPrimitive || this.instructions;
+    const instructions = typeof instructionStringOrFunction === 'function' ? instructionStringOrFunction(runtimeContext) : instructionStringOrFunction;
```

Comment:
It should be an object so if we want to add more args we can.

```suggestion
    const instructions = typeof instructionStringOrFunction === 'function' ? instructionStringOrFunction({ runtimeContext: runtimeContext ?? new Container() }) : instructionStringOrFunction;
```